src/seedu/addressbook/data/person/Phone.java

```diff
@@ -21,12 +21,14 @@
      * @throws IllegalValueException if given phone string is invalid.
      */
     public Phone(String phone, boolean isPrivate) throws IllegalValueException {
+    	super (phone, isPrivate);
         this.isPrivate = isPrivate;
         phone = phone.trim();
         if (!isValidPhone(phone)) {
             throw new IllegalValueException(MESSAGE_PHONE_CONSTRAINTS);
         }
```

Comment:
validating inputs should be a part of the UI level. Think when/ when not to use exceptions